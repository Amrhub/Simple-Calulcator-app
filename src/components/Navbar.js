import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <nav className="navbar flex-sb-center">
    <span className="navbar-brand">math magicians</span>

    <ul className="navbar-nav">
      <li>
        <NavLink
          exact
          activeClassName="active-link"
          to="/"
          className="nav-link"
        >
          Home
        </NavLink>
        <NavLink
          exact
          activeClassName="active-link"
          to="/calculator"
          className="nav-link"
        >
          Calculator
        </NavLink>
        <NavLink
          exact
          activeClassName="active-link"
          to="/quote"
          className="nav-link"
        >
          Quote
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
