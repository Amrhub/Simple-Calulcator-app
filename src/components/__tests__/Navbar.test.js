// eslint-disable-next-line no-unused-vars
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../Navbar';

it('Navbar component renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
