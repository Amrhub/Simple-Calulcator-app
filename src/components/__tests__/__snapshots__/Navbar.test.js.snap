// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<nav
  className="navbar flex-sb-center"
>
  <span
    className="navbar-brand"
  >
    math magicians
  </span>
  <ul
    className="navbar-nav"
  >
    <li>
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        className="nav-link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
